core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -638,6 +639,13 @@ public static boolean eq(Object b0, Object b1) {
     return b0.equals(b1);
   }
 
+  /** SQL <code>=</code> operator applied to Object values (including String;
+   * neither side may be null). */
+  public static boolean eq(Object b0, Object b1, Comparator comparator) {
```

Comment:
Normally we should call this method only with parameters of type `String`. Why do we need `Object`?

Reply:
Parameters modified.